server/src/main/java/io/druid/metadata/SQLMetadataConnector.java

```diff
@@ -369,4 +377,37 @@ protected BasicDataSource getDatasource()
     return dataSource;
   }
 
+  private void createAuditTable(final IDBI dbi, final String tableName)
+  {
+    createTable(
+        dbi,
+        tableName,
+        ImmutableList.of(
+            String.format(
+                "CREATE TABLE %1$s (\n"
```

Comment:
I think we need a blob in here about what the previous state or something was. E.g., if we modified a rule, what was the previous rule?